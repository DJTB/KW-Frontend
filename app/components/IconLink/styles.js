import styled from 'styled-components';

import A from 'base/A';
import { ghost } from 'shared/styles/utils';
import { fastEaseQuad } from 'shared/styles/animation';

export const Link = styled(A)`
  transition: all ${fastEaseQuad}, transform 100ms linear;
  cursor: pointer;
  opacity: .7;
  ${({ visuallyHidden }) => visuallyHidden && ghost}

  &:focus,
  &:hover {
    opacity: 1;
    outline: none;
  }

  &:active {
    opacity: 1;
  }

  & > * {
    transform: scale(1);
    &:active {
      transform: scale(.9);
    }
  }
`;
