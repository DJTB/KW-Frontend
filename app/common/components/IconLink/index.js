import React from "react";
import PropTypes from "prop-types";

import Icon from "common/components/Icon";
import { Link } from "./styles";

// color and size have defaults in <Icon /> already
/* eslint-disable react/require-default-props */
IconLink.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  href: PropTypes.string,
  to: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};
/* eslint-enable */

IconLink.defaultProps = {
  href: "",
  to: "",
};

function IconLink({
  name, title, color, size, href, to, ...props
}) {
  return (
    <Link title={title} href={href} to={to} {...props}>
      <Icon inline={false} name={name} color={color} size={size} />
    </Link>
  );
}

export default IconLink;
