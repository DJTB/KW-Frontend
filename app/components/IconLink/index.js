import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import A from 'base/A';
import Icon from 'components/Icon';
import { fastEaseQuad } from 'shared/styles/animation';

const StyledAnchor = styled(A)`
  transition: all ${fastEaseQuad}, transform 100ms linear;
  cursor: pointer;
  opacity: .7;

  &:focus,
  &:hover {
    opacity: .9;
    outline: none;
  }

  &:active {
    opacity: 1;
    transform: scale(.9);
  }
`;

IconLink.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  href: PropTypes.string,
  to: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

IconLink.defaultProps = {
  color: 'currentColor',
  size: '1em',
  href: '',
  to: '',
};

function IconLink({ name, title, color, size, href, to, ...props }) {
  return (
    <StyledAnchor title={title} href={href} to={to} {...props} >
      <Icon
        display="block"
        name={name}
        color={color}
        size={size}
      />
    </StyledAnchor>
  );
}


export default IconLink;
