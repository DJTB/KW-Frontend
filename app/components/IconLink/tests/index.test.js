import React from 'react';
import { render } from 'enzyme';
import IconLink from '../index';

describe('<IconLink /> ', () => {
  it('should match baseline snapshot of anchor', () => {
    const renderedComponent = render(
      <IconLink name="ADD" title="Does an action" href="http://google.com" />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should match baseline snapshot of react-router Link', () => {
    const renderedComponent = render(
      <IconLink name="ADD" title="Does an action" to="/path" />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
