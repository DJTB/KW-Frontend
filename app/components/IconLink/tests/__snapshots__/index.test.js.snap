// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconLink />  should match baseline snapshot of anchor 1`] = `
<a
  class="kjCPoV iLbNOi"
  href="http://google.com"
  title="Does an action"
>
  <span
    class="hNDMaS"
    color="currentColor"
    display="block"
  >
    <svg
      class="hfAVjD"
      height="100%"
      preserveaspectratio="xMidYMid meet"
      title="ADD"
      viewbox="0 0 24 24"
      width="100%"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
      />
    </svg>
  </span>
</a>
`;

exports[`<IconLink />  should match baseline snapshot of react-router Link 1`] = `
<a
  class="kjCPoV iLbNOi"
  title="Does an action"
>
  <span
    class="hNDMaS"
    color="currentColor"
    display="block"
  >
    <svg
      class="hfAVjD"
      height="100%"
      preserveaspectratio="xMidYMid meet"
      title="ADD"
      viewbox="0 0 24 24"
      width="100%"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
      />
    </svg>
  </span>
</a>
`;
